Antrea agent version v0.7.0 is crashing in EKS env

```
I0604 09:52:03.152705       1 agent.go:53] Starting Antrea agent (version v0.7.0)
I0604 09:52:03.153138       1 client.go:34] No kubeconfig file was specified. Falling back to in-cluster config
W0604 09:52:03.154078       1 configmap_cafile_content.go:102] unable to load initial CA bundle for: "antrea-ca::kube-system::antrea-ca::ca.crt" due to: configmap "antrea-ca" not found
I0604 09:52:03.158268       1 ovs_client.go:64] Connecting to OVSDB at address /var/run/openvswitch/db.sock
I0604 09:52:03.161105       1 agent.go:177] Setting up node network
I0604 09:52:03.171547       1 ovs_client.go:105] Bridge exists: f212da14-36eb-443e-96d5-e37fa4b0945b
I0604 09:52:03.172197       1 agent.go:582] No round number found in OVSDB, using 1
I0604 09:52:03.172207       1 agent.go:594] Using round number 1
I0604 09:52:03.172248       1 ofctrl.go:170] Initialize connection or re-connect to /var/run/openvswitch/br-int.mgmt.
I0604 09:52:04.172602       1 ofctrl.go:185] Connected to socket /var/run/openvswitch/br-int.mgmt
I0604 09:52:04.172744       1 ofctrl.go:247] New connection..
I0604 09:52:04.172765       1 ofctrl.go:254] Send hello with OF version: 4
I0604 09:52:04.173061       1 ofctrl.go:268] Received Openflow 1.3 Hello message
I0604 09:52:04.173462       1 ofctrl.go:285] Received ofp1.3 Switch feature response: {Header:{Version:4 Type:6 Length:32 Xid:3} DPID:00:00:16:da:12:f2:3e:44 Buffers:0 NumTables:254 AuxilaryId:0 pad:[0 0] Capabilities:79 Actions:0 Ports:[]}
I0604 09:52:04.173492       1 ofSwitch.go:84] Openflow Connection for new switch: 00:00:16:da:12:f2:3e:44
I0604 09:52:04.173890       1 ofctrl_bridge.go:207] OFSwitch is connected: 00:00:16:da:12:f2:3e:44
I0604 09:52:04.197332       1 agent.go:206] Agent initialized NodeConfig=NodeName: ip-192-168-38-63.ec2.internal, OVSBridge: , PodCIDR: <nil>, NodeIP: 192.168.38.63/19, Gateway: Name gw0: IP 192.168.38.63, MAC 62:0f:de:84:fb:97, NetworkConfig=&{NetworkPolicyOnly vxlan false }
I0604 09:52:04.197484       1 server.go:616] Reconciliation for CNI server
I0604 09:52:04.209301       1 agent.go:347] Cleaning up flow-restore-wait config
I0604 09:52:04.210219       1 agent.go:351] Cleaned up flow-restore-wait config
W0604 09:52:04.210244       1 client.go:97] Didn't get CA certificate from ConfigMap. May not be able to verify server cert
I0604 09:52:04.210250       1 client.go:102] No antrea kubeconfig file was specified. Falling back to in-cluster config
I0604 09:52:04.210398       1 client.go:122] Updating Antrea client with the new CA bundle
I0604 09:52:04.215126       1 server.go:551] Starting CNI server
I0604 09:52:04.215402       1 server.go:561] CNI server is listening ...
I0604 09:52:04.219192       1 configmap_cafile_content.go:205] Starting antrea-ca::kube-system::antrea-ca::ca.crt
I0604 09:52:04.219208       1 shared_informer.go:197] Waiting for caches to sync for antrea-ca::kube-system::antrea-ca::ca.crt
I0604 09:52:04.219263       1 agent.go:45] Starting Antrea Agent Monitor
I0604 09:52:04.219667       1 networkpolicy_controller.go:400] Starting watch for AppliedToGroup
I0604 09:52:04.219920       1 networkpolicy_controller.go:400] Starting watch for AddressGroup
I0604 09:52:04.220095       1 networkpolicy_controller.go:400] Starting watch for NetworkPolicy
W0604 09:52:04.226668       1 networkpolicy_controller.go:403] Failed to start watch for NetworkPolicy: Get https://10.100.153.206:443/apis/networking.antrea.tanzu.vmware.com/v1beta1/networkpolicies?fieldSelector=nodeName%3Dip-192-168-38-63.ec2.internal&watch=true: x509: certificate signed by unknown authority
W0604 09:52:04.227171       1 networkpolicy_controller.go:403] Failed to start watch for AppliedToGroup: Get https://10.100.153.206:443/apis/networking.antrea.tanzu.vmware.com/v1beta1/appliedtogroups?fieldSelector=nodeName%3Dip-192-168-38-63.ec2.internal&watch=true: x509: certificate signed by unknown authority
E0604 09:52:04.228820       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 208 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1a1c400, 0x2e00360)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1a1c400, 0x2e00360)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
net.networkNumberAndMask(0x0, 0xb, 0x12, 0xc000056009, 0x1ca1d58, 0x3, 0xc00005602e)
	/usr/local/go/src/net/ip.go:473 +0x26
net.(*IPNet).String(0x0, 0xc00005403a, 0x1d)
	/usr/local/go/src/net/ip.go:523 +0x40
github.com/vmware-tanzu/antrea/pkg/agent/querier.agentQuerier.GetAgentInfo(0xc000240500, 0x2001f60, 0xc0000c8320, 0x201edc0, 0xc0002d2dc0, 0x20108a0, 0xc000316120, 0x200fa40, 0xc0002f4080, 0x1ff7440, ...)
	/antrea/pkg/agent/querier/querier.go:196 +0x2ba
github.com/vmware-tanzu/antrea/pkg/monitor.(*agentMonitor).createAgentCRD(0xc0007d63f0, 0xc00023cdc0, 0xc0004aa698, 0x8)
	/antrea/pkg/monitor/agent.go:96 +0x57
github.com/vmware-tanzu/antrea/pkg/monitor.(*agentMonitor).syncAgentCRD(0xc0007d63f0)
	/antrea/pkg/monitor/agent.go:64 +0x274
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0006a57a0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0006d1fa0, 0xdf8475800, 0x0, 0xc0006a5701, 0xc00074dda0)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/go/pkg/mod/k8s.io/apimachinery@v0.17.6/pkg/util/wait/wait.go:88
github.com/vmware-tanzu/antrea/pkg/monitor.(*agentMonitor).Run(0xc0007d63f0, 0xc00074dda0)
	/antrea/pkg/monitor/agent.go:48 +0xd2
created by main.run
	/antrea/cmd/antrea-agent/agent.go:182 +0x11a8
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x521046]
```

--- a/pkg/agent/agent.go
+++ b/pkg/agent/agent.go
@@ -503,7 +503,11 @@ func (i *Initializer) initNodeLocalConfig() error {
 	}
 
 	if i.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
-		i.nodeConfig = &config.NodeConfig{Name: nodeName, NodeIPAddr: localAddr, BridgeName: i.ovsBridgeClient.GetBridgeName(), UplinkNetConfig: new(config.AdapterNetConfig)}
+		i.nodeConfig = &config.NodeConfig{
+			Name:            nodeName,
+			OVSBridge:       i.ovsBridge,
+			NodeIPAddr:      localAddr,
+			UplinkNetConfig: new(config.AdapterNetConfig)}
 		return nil
 	}
 
@@ -524,7 +528,6 @@ func (i *Initializer) initNodeLocalConfig() error {
 		OVSBridge:       i.ovsBridge,
 		PodCIDR:         localSubnet,
 		NodeIPAddr:      localAddr,
-		BridgeName:      i.ovsBridgeClient.GetBridgeName(),
 		UplinkNetConfig: new(config.AdapterNetConfig)}
 	return nil
 }

--- a/pkg/agent/config/node_config.go
+++ b/pkg/agent/config/node_config.go
@@ -58,12 +58,17 @@ type AdapterNetConfig struct {
 
 // Local Node configurations retrieved from K8s API or host networking state.
 type NodeConfig struct {
-	Name            string
-	OVSBridge       string
-	PodCIDR         *net.IPNet
-	NodeIPAddr      *net.IPNet
+	// The Node's name used in Kubernetes.
+	Name string
+	// The name of the OpenVSwitch bridge antrea-agent uses.
+	OVSBridge string
+	// The CIDR block to allocate Pod IPs out of.
+	// It's nil for the networkPolicyOnly trafficEncapMode which doesn't do IPAM.
+	PodCIDR *net.IPNet
+	// The Node's IP used in Kubernetes. It has the network mask information.
+	NodeIPAddr *net.IPNet
+	// The config of the gateway network device, i.e. gw0 by default.
 	GatewayConfig   *GatewayConfig
-	BridgeName      string
 	UplinkNetConfig *AdapterNetConfig
 }
 

--- a/pkg/agent/querier/querier.go
+++ b/pkg/agent/querier/querier.go
@@ -193,7 +193,9 @@ func (aq agentQuerier) GetAgentInfo(agentInfo *v1beta1.AntreaAgentInfo, partial 
 		agentInfo.Version = querier.GetVersion()
 		agentInfo.PodRef = querier.GetSelfPod()
 		agentInfo.NodeRef = querier.GetSelfNode(true, aq.nodeConfig.Name)
-		agentInfo.NodeSubnet = []string{aq.nodeConfig.PodCIDR.String()}
+		if aq.nodeConfig.PodCIDR != nil {
+			agentInfo.NodeSubnet = []string{aq.nodeConfig.PodCIDR.String()}
+		}
 		agentInfo.OVSInfo.BridgeName = aq.nodeConfig.OVSBridge
 		agentInfo.APIPort = aq.apiPort
 	}

--- a/pkg/agent/querier/querier_test.go
+++ b/pkg/agent/querier/querier_test.go
@@ -1,0 +1,192 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package querier
+
+import (
+	"net"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/vmware-tanzu/antrea/pkg/agent/config"
+	interfacestoretest "github.com/vmware-tanzu/antrea/pkg/agent/interfacestore/testing"
+	openflowtest "github.com/vmware-tanzu/antrea/pkg/agent/openflow/testing"
+	"github.com/vmware-tanzu/antrea/pkg/apis/clusterinformation/v1beta1"
+	binding "github.com/vmware-tanzu/antrea/pkg/ovs/openflow"
+	ovsconfigtest "github.com/vmware-tanzu/antrea/pkg/ovs/ovsconfig/testing"
+	queriertest "github.com/vmware-tanzu/antrea/pkg/querier/testing"
+)
+
+const ovsVersion = "2.10.0"
+
+func getIPNet(ip string) *net.IPNet {
+	_, ipNet, _ := net.ParseCIDR(ip)
+	return ipNet
+}
+
+func TestAgentQuerierGetAgentInfo(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	interfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	interfaceStore.EXPECT().GetContainerInterfaceNum().Return(2).AnyTimes()
+
+	ofClient := openflowtest.NewMockClient(ctrl)
+	ofClient.EXPECT().GetFlowTableStatus().Return([]binding.TableStatus{
+		{
+			ID:        1,
+			FlowCount: 2,
+		},
+	}).AnyTimes()
+	ofClient.EXPECT().IsConnected().Return(true).AnyTimes()
+
+	ovsBridgeClient := ovsconfigtest.NewMockOVSBridgeClient(ctrl)
+	ovsBridgeClient.EXPECT().GetOVSVersion().Return(ovsVersion, nil).AnyTimes()
+
+	networkPolicyInfoQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+	networkPolicyInfoQuerier.EXPECT().GetNetworkPolicyNum().Return(10).AnyTimes()
+	networkPolicyInfoQuerier.EXPECT().GetAppliedToGroupNum().Return(20).AnyTimes()
+	networkPolicyInfoQuerier.EXPECT().GetAddressGroupNum().Return(30).AnyTimes()
+	networkPolicyInfoQuerier.EXPECT().GetControllerConnectionStatus().Return(true).AnyTimes()
+
+	tests := []struct {
+		name              string
+		nodeConfig        *config.NodeConfig
+		apiPort           int
+		partial           bool
+		expectedAgentInfo *v1beta1.AntreaAgentInfo
+	}{
+		{
+			name: "networkPolicyOnly-mode non-partial",
+			nodeConfig: &config.NodeConfig{
+				Name:       "foo",
+				OVSBridge:  "br-int",
+				NodeIPAddr: getIPNet("10.10.0.10"),
+			},
+			apiPort: 10350,
+			partial: false,
+			expectedAgentInfo: &v1beta1.AntreaAgentInfo{
+				ObjectMeta: v1.ObjectMeta{Name: "foo"},
+				NodeRef:    corev1.ObjectReference{Kind: "Node", Name: "foo"},
+				NodeSubnet: nil,
+				OVSInfo: v1beta1.OVSInfo{
+					Version:    ovsVersion,
+					BridgeName: "br-int",
+					FlowTable:  map[string]int32{"1": 2},
+				},
+				NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+					NetworkPolicyNum:  10,
+					AppliedToGroupNum: 20,
+					AddressGroupNum:   30,
+				},
+				LocalPodNum: 2,
+				AgentConditions: []v1beta1.AgentCondition{
+					{
+						Type:   v1beta1.AgentHealthy,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.ControllerConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.OVSDBConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.OpenflowConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+				},
+				APIPort: 10350,
+				Version: "UNKNOWN",
+			},
+		},
+		{
+			name: "encap-mode non-partial",
+			nodeConfig: &config.NodeConfig{
+				Name:       "foo",
+				OVSBridge:  "br-int",
+				NodeIPAddr: getIPNet("10.10.0.10"),
+				PodCIDR:    getIPNet("20.20.20.0/24"),
+			},
+			apiPort: 10350,
+			partial: false,
+			expectedAgentInfo: &v1beta1.AntreaAgentInfo{
+				ObjectMeta: v1.ObjectMeta{Name: "foo"},
+				NodeRef:    corev1.ObjectReference{Kind: "Node", Name: "foo"},
+				NodeSubnet: []string{"20.20.20.0/24"},
+				OVSInfo: v1beta1.OVSInfo{
+					Version:    ovsVersion,
+					BridgeName: "br-int",
+					FlowTable:  map[string]int32{"1": 2},
+				},
+				NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+					NetworkPolicyNum:  10,
+					AppliedToGroupNum: 20,
+					AddressGroupNum:   30,
+				},
+				LocalPodNum: 2,
+				AgentConditions: []v1beta1.AgentCondition{
+					{
+						Type:   v1beta1.AgentHealthy,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.ControllerConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.OVSDBConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+					{
+						Type:   v1beta1.OpenflowConnectionUp,
+						Status: corev1.ConditionTrue,
+					},
+				},
+				APIPort: 10350,
+				Version: "UNKNOWN",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			aq := agentQuerier{
+				nodeConfig:               tt.nodeConfig,
+				interfaceStore:           interfaceStore,
+				ofClient:                 ofClient,
+				ovsBridgeClient:          ovsBridgeClient,
+				networkPolicyInfoQuerier: networkPolicyInfoQuerier,
+				apiPort:                  tt.apiPort,
+			}
+			agentInfo := &v1beta1.AntreaAgentInfo{}
+			aq.GetAgentInfo(agentInfo, tt.partial)
+			// Check AgentConditions separately as it contains timestamp we cannot predict.
+			assert.Equal(t, len(tt.expectedAgentInfo.AgentConditions), len(agentInfo.AgentConditions))
+			for i := range agentInfo.AgentConditions {
+				assert.Equal(t, tt.expectedAgentInfo.AgentConditions[i].Status, agentInfo.AgentConditions[i].Status)
+				assert.Equal(t, tt.expectedAgentInfo.AgentConditions[i].Type, agentInfo.AgentConditions[i].Type)
+			}
+			// Exclude AgentConditions before comparing the whole objects.
+			tt.expectedAgentInfo.AgentConditions = nil
+			agentInfo.AgentConditions = nil
+			assert.Equal(t, tt.expectedAgentInfo, agentInfo)
+		})
+	}
+}

--- a/pkg/version/version.go
+++ b/pkg/version/version.go
@@ -49,7 +49,7 @@ func GetGitSHA() string {
 // unreleased versions.
 func GetFullVersion() string {
 	if Version == "" {
-		return "UKNOWN"
+		return "UNKNOWN"
 	}
 	if ReleaseStatus == "released" {
 		return Version

